pkg/istio-agent/xds_proxy_delta.go

```diff
@@ -162,7 +162,7 @@ func (p *XdsProxy) handleDeltaUpstream(ctx context.Context, con *ProxyConnection
 		select {
 		case err := <-con.upstreamError:
 			// error from upstream Istiod.
-			if istiogrpc.IsExpectedGRPCError(err) {
+			if istiogrpc.IsExpectedGRPCError(err) || isRateLimited(err) {
```

Comment:
considers rate limiting as expected error

Reply:
I am not sure if this is right. In this case, we should let users know what happen, instead of hide it

Reply:
But Envoy reconnects like any other gRPC problem like Istiod disconnects every 30 mins. I do not know how is different from that. Istiod disconnects every 30 mins to shed the load.

Reply:
See https://github.com/istio/istio/pull/40644#issuecomment-1226887934